.github/actions/setup-sui/action.yaml

```diff
@@ -0,0 +1,67 @@
+name: Setup SUI Cli
+description: 'Setup Sui CLI and install dependencies'
+
+inputs:
+  SUI_VERSION:
+    description: 'The version of Sui CLI to install'
+    required: true
+
+runs:
+  using: 'composite'
+
+  steps:
+    - name: Debug Action Input
+      shell: bash
+      run: echo "SUI_VERSION=${{ inputs.SUI_VERSION }}"
+
+    - name: Install Dependencies
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libpq-dev
+
+    - name: Cache Sui binaries
+      id: cache-sui
+      uses: actions/cache@v4
+      with:
+        path: sui-cli/sui-binaries/
+        key: sui-${{ inputs.SUI_VERSION }}
+
+    - name: Download and Install Sui
```

Comment:
Can we import the action from axelar-cgp-sui instead to avoid duplication?

Reply:
I need some time research and experiment around it. let me work on this in a separate pr; this shouldn't be a blocker for this PR to merge.